test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -557,12 +575,159 @@ public void AnomalyDetectionWithSrCnn(bool loadDataFromFile)
             int k = 0;
             foreach (var prediction in predictionColumn)
             {
+                Assert.Equal(3, prediction.Prediction.Length);
                 if (k == 20)
                     Assert.Equal(1, prediction.Prediction[0]);
                 else
                     Assert.Equal(0, prediction.Prediction[0]);
                 k += 1;
             }
         }
+
+        [Theory]
+        [InlineData(SrCnnDetectMode.AnomalyOnly)]
+        [InlineData(SrCnnDetectMode.AnomalyAndExpectedValue)]
+        [InlineData(SrCnnDetectMode.AnomalyAndMargin)]
+        [InlineData(SrCnnDetectMode.AnomalyOnly, true)]
+        [InlineData(SrCnnDetectMode.AnomalyAndExpectedValue, true)]
+        [InlineData(SrCnnDetectMode.AnomalyAndMargin, true)]
+        public void AnomalyDetectionWithSrCnnEntire(SrCnnDetectMode mode, bool loadDataFromFile=false)
```

Comment:
You can replace all this with just [CombinatorialData]
See [https://github.com/AArnott/Xunit.Combinatorial](https://github.com/AArnott/Xunit.Combinatorial)